### PR TITLE
net-misc/seafile-client: bump including dependencies

### PR DESCRIPTION
**Pull request by dkriegner**
fixes problems with Qt 5.11
Bug: https://bugs.gentoo.org/661408

### CODE REVIEW

**Comment by dkriegner** [comment]
@mgorny, you got me, I did not do my due diligence

Here what I did now:

- I split the original commit into one commit per package
- diffs on the build files showed that both seafile and seafile-client dropped the dependency to net-libs/ccnet
- net-misc/seafile was rather trivially to bumped to EAPI7
- net-misc/seafile-client compiles with and without the shibboleth use flag. (I can't test the shibboleth login itself)
- net-misc/seafile-client performs file synchronization as intended
- net-misc/seafile-client: the patch `seafile-client-select-qt5.patch` is not needed for me to compile and run this. @moschlar should confirm its still needed.

Note that after these packages are bumbed I think there is no package left in the tree with dependency on net-libs/ccnet

**Comment by mgorny** [comment]
Thanks a lot. Let's wait a few days for @moschlar to review. If he doesn't reply, please ping me again.

**Comment by moschlar** [comment]
Pong! Thanks for the bump, I'm just checking out your changes.

**Comment by moschlar** [comment]
Thanks, @dkriegner !

> - diffs on the build files showed that both seafile and seafile-client dropped the dependency to net-libs/ccnet

Yeah, they dropped the dependency from their build files, but at least in seafile, the ccnet Python module is still required by the `seaf-cli` script... I've already complained to them that their dependency removal process has not been done properly and isn't finished and they said they were still working on it. Just using the GUI part in seafile-client works with or without ccnet installed though. But for the meantime, I would prefer to keep the ccnet dependency because I really want `seaf-cli` to stay usable. (If you re-add it, please demote it to `~net-libs/ccnet-6.1.7`, dunno why I hadn't done this before.

> - net-misc/seafile-client: the patch seafile-client-select-qt5.patch is not needed for me to compile and run this. @moschlar should confirm its still needed.

I'm actually not sure anymore if it was technically required or just because it's the proper way of querying the Qt version (I tried to upstream it, but well https://github.com/haiwen/seafile-client/pull/918).
It doesn't do any harm now, does it?

> - net-misc/seafile-client-6.2.4 now depends on ~net-misc/seafile-${PV} which seems ok, but more then required by upstream which in the debian package requires only a lower version

I don't think that that dependency information there is reliable... Their upstream packaging guide strongly advises to always use the same versions of all three/two packages.

> Note that after these packages are bumbed I think there is no package left in the tree with dependency on net-libs/ccnet

Well, at the moment, there still is. I'll bug upstream again regarding the `seaf-cli` problem.

So to summarize: I'd prefer having the ccnet dependency added again but apart from that, go ahead!

**Comment by dkriegner** [comment]
I agree with the comments of @moschlar 
I added back the dependency on ccnet. I left it at version 6.1.7 since 6.1.8 refers to the same commit anyways. 

I hope it can be merged as is now.

**Comment by moschlar** [comment]
👍

Moritz Schlarb (Mobil)

**Comment by mgorny** [comment]
Thanks to you both! This is really good work, I'm merging it now.
